### ISSUE TITLE
Environment reset

### ISSUE THREAD

**Issue by b-fg** [flexi-framework/relexi#5]
Hello! I was wondering what does the `reset` method call in the environment definition:
https://github.com/flexi-framework/relexi/blob/7b8e0dad8e9f36af4c18fd6058e07a3864c758ee/src/relexi/env/flexiEnvSmartSim.py#L291 while only the `_reset` method is defined:
https://github.com/flexi-framework/relexi/blob/7b8e0dad8e9f36af4c18fd6058e07a3864c758ee/src/relexi/env/flexiEnvSmartSim.py#L264

Is this a bug or intended? If the latter, could you please explain?
Thanks!

**Comment by m-kurz** [comment]
Hi! This is not a bug, but stems from how custom environments are implemented in TF-Agents. The FLEXI environment subclasses from the standard `PyEnvironment` class provided by TF-Agents and has to implement some specific methods. To cite the documentation on the `reset()` method:
> Note: Subclasses cannot override this directly. Subclasses implement
> _reset() which will be called by this method. The output of _reset() will
> be cached and made available through current_time_step().

You can also have a look at the code on Github:
-  [https://github.com/tensorflow/agents/blob/v0.16.0/tf_agents/environments/py_environment.py#L180-L196](https://github.com/tensorflow/agents/blob/v0.16.0/tf_agents/environments/py_environment.py#L180-L196)

or the documentation:
- [https://www.tensorflow.org/agents/api_docs/python/tf_agents/environments/PyEnvironment#reset](https://www.tensorflow.org/agents/api_docs/python/tf_agents/environments/PyEnvironment#reset)

Hope this helps!

**TL;DR**: The `reset()` method is implemented by the standard `PyEnvironment` we subclass from, and this `reset()` method just calls the `_reset()` method that has to be implement by our custom environment.

**Comment by b-fg** [comment]
Noted, thanks!
